Source/Core/Common/MsgHandler.cpp

```diff
@@ -83,35 +83,27 @@ std::string GetStringT(const char* string)
 bool MsgAlert(bool yes_no, MsgType style, const char* format, ...)
 {
   // Read message and write it to the log
-  std::string caption;
+  const char* caption = "";
   char buffer[2048];
 
-  static std::string info_caption;
-  static std::string warn_caption;
-  static std::string ques_caption;
-  static std::string crit_caption;
-
-  if (info_caption.empty())
-  {
-    info_caption = s_str_translator(_trans("Information"));
-    ques_caption = s_str_translator(_trans("Question"));
-    warn_caption = s_str_translator(_trans("Warning"));
-    crit_caption = s_str_translator(_trans("Critical"));
-  }
+  static const std::string info_caption = s_str_translator(_trans("Information"));
+  static const std::string warn_caption = s_str_translator(_trans("Question"));
+  static const std::string ques_caption = s_str_translator(_trans("Warning"));
+  static const std::string crit_caption = s_str_translator(_trans("Critical"));
```

Comment:
Is there any reason to call `s_str_translator` for all four strings each time we run `MsgAlert`, or could we put each `s_str_translator` call in one of the switch cases? (Maybe that would require undoing the `std::string` vs `char*` changes you did, though...)

Reply:
They will not be instantiated every time, since they are static strings. Instead, they'll be instantiated once in a **thread safe** manner - unlike the old code. See:
https://en.cppreference.com/w/cpp/language/storage_duration#Static_local_variables

Reply:
Right, sorry, I missed the `static` :)